autoload/vital/__vital__/VS/Vim/Window.vim

```diff
@@ -30,14 +30,97 @@ function! s:do(winid, func) abort
   noautocmd keepalt keepjumps call win_gotoid(l:curr_winid)
 endfunction
 
+"
+" scrollable
+"
+function! s:scrollable(win, ...) abort
+  if len(a:000) == 1
+    call s:set_var(a:win, 'scrollable', a:000[0])
+  else
+    return s:get_var(a:win, 'scrollable')
+  endif
+endfunction
+
+"
+" set_var
+"
+function! s:set_var(win, key, value) abort
+  if winheight(a:win) != -1
+    let l:map = getwinvar(a:win, '___VS_Vim_Window', {})
+    let l:map[a:key] = a:value
+    call setwinvar(a:win, '___VS_Vim_Window', l:map)
+  endif
+endfunction
+
+"
+" get_var
+"
+function! s:get_var(win, key) abort
+  if winheight(a:win) != -1
+    let l:map = getwinvar(a:win, '___VS_Vim_Window', {})
+    return get(l:map, a:key, v:null)
+  endif
+  return v:null
+endfunction
+
+"
+" scroll
+"
+" NOTE: We can't test it because it uses timer.
+"
+function! s:scroll(delta) abort
+  if a:delta == 0
+    return
+  endif
+
+  let l:wins = []
+  let l:wins += map(range(1, tabpagewinnr(tabpagenr(), '$')), 'win_getid(v:val)')
+  let l:wins += exists('*popup_list') ? popup_list() : []
+  for l:win in l:wins
+    if s:scrollable(l:win)
+      let l:ctx = {}
+      function! l:ctx.callback(win, delta) abort
+        let l:height = line('w$') - line('w0')
```

Comment:
since this is a callback due to timer, the window could be deleted. should there be some guard on win exist?

Reply:
Agree. I will add it.